### PR TITLE
Only use p2 workflow on staging.

### PR DESCRIPTION
**Pull request by pgk**
Closes https://github.com/Automattic/greenhouse/issues/2087

## Proposed Changes

* Update P2 signup flow to only be available on staging, which means it will only work for proxied a11ns according to FG.
* In other cases, should it redirect to some other flow?

## Testing Instructions

* Running `http://calypso.local/start/p2` should not get you the P2 signup flow.
* by setting the config value for development to true, you get the flow.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by gikaragia — APPROVED** [review]
Had some issues with the calypso enviornment but managed to make it work! LGTM!

**Comment by jgcaruso** [comment]
Should you set `p2-enabled` explicitely to `true` for the staging config, just so it doesn't look like its missing? I didn't notice it in the edited config files.

File: config/stage.json [hunk 1]
```diff
@@ -120,6 +120,7 @@
 		"onboarding/new-user-survey": false,
 		"onboarding/new-user-survey-scrambled": false,
 		"p2/p2-plus": true,
+		"p2-enabled": false,
```
Comment (pgk) [inline config/stage.json]:
```suggestion
		"p2-enabled": true,
```

**Comment by pgk** [comment]
> Should you set `p2-enabled` explicitely to `true` for the staging config, just so it doesn't look like its missing? I didn't notice it in the edited config files.

Yes, good catch!

**Pull request merged**